includes/AMP/Output_Buffer.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Class Output_Buffer
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\AMP;
+
+use DOMElement;
+use Exception;
+use Google\Web_Stories\Infrastructure\Conditional;
+use Google\Web_Stories\Service_Base;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Traits\Publisher;
+use Google\Web_Stories_Dependencies\AmpProject\Dom\Document;
+use Throwable;
+
+/**
+ * Output buffer class.
+ *
+ * Largely copied from AMP_Theme_Support.
+ *
+ * @since 1.10.0
+ *
+ * @see \AMP_Theme_Support
+ */
+class Output_Buffer extends Service_Base implements Conditional {
+	use Publisher;
+
+	/**
+	 * Whether output buffering has started.
+	 *
+	 * @var bool
+	 */
+	protected $is_output_buffering = false;
+
+	/**
+	 * Sanitization instance.
+	 *
+	 * @var Sanitization Sanitization instance.
+	 */
+	private $sanitization;
+
+	/**
+	 * Optimization instance.
+	 *
+	 * @var Optimization Optimization instance.
+	 */
+	private $optimization;
+
+	/**
+	 * Output_Buffer constructor.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @param Sanitization $sanitization Sanitization instance.
+	 * @param Optimization $optimization Optimization instance.
+	 */
+	public function __construct( Sanitization $sanitization, Optimization $optimization ) {
+		$this->sanitization = $sanitization;
+		$this->optimization = $optimization;
+	}
+
+	/**
+	 * Runs on instantiation.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @return void
+	 */
+	public function register() {
+		/*
+		 * Start output buffering at very low priority for sake of plugins and themes that use template_redirect
+		 * instead of template_include.
+		 */
+		$priority = defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX; // phpcs:ignore PHPCompatibility.Constants.NewConstants.php_int_minFound
+		add_action( 'template_redirect', [ $this, 'start_output_buffering' ], $priority );
+	}
+
```

Comment:
```suggestion
	public function register() {
		/*
		 * Start output buffering at very low priority for sake of plugins and themes that use template_redirect
		 * instead of template_include.
		 */
		$this->start_output_buffering();
	}
    
    /**
	 * Get the action to use for registering the service.
	 *
	 * @since 1.6.0
	 *
	 * @return string Registration action to use.
	 */
	public static function get_registration_action(): string {
		return 'template_redirect';
	}

	/**
	 * Get the action priority to use for registering the service.
	 *
	 * @since 1.6.0
	 *
	 * @return int Registration action priority to use.
	 */
	public static function get_registration_action_priority(): int {
	    $priority = defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX; // phpcs:ignore PHPCompatibility.Constants.NewConstants.php_int_minFound
		return $priority;
	}
```

How about this?